src/transformers/pipelines.py

```diff
@@ -553,6 +553,232 @@ def __call__(self, *args, **kwargs):
+    # in https://github.com/rusiaaman/XLNet-gen#methodology
+    # and https://medium.com/@amanrusia/xlnet-speaks-comparison-to-gpt-2-ea1a4e9ba39e
+    PADDING_TEXT = """In 1991, the remains of Russian Tsar Nicholas II and his family
+    (except for Alexei and Maria) are discovered.
+    The voice of Nicholas's young son, Tsarevich Alexei Nikolaevich, narrates the
+    remainder of the story. 1883 Western Siberia,
+    a young Grigori Rasputin is asked by his father and a group of men to perform magic.
+    Rasputin has a vision and denounces one of the men as a horse thief. Although his
+    father initially slaps him for making such an accusation, Rasputin watches as the
+    man is chased outside and beaten. Twenty years later, Rasputin sees a vision of
+    the Virgin Mary, prompting him to become a priest. Rasputin quickly becomes famous,
+    with people, even a bishop, begging for his blessing. <eod> </s> <eos>"""
+
+    def __init__(
+        self,
+        model: Union["PreTrainedModel", "TFPreTrainedModel"],
+        tokenizer: PreTrainedTokenizer,
+        modelcard: Optional[ModelCard] = None,
+        framework: Optional[str] = None,
+        args_parser: ArgumentHandler = None,
+        device: int = -1,
+        task: str = "",
+        length: int = 20,
+        temperature: float = 1.0,
+        top_k: int = 0,
+        top_p: float = 0.9,
+        repetition_penalty: float = 1.0,
+        do_sample: bool = True,
+        num_return_sequences: int = 1,
+        stop_token: str = None,
+        padding_text: str = "",
+        xlm_language: str = "en",
+    ):
+        super().__init__(
+            model=model,
+            tokenizer=tokenizer,
+            modelcard=modelcard,
+            framework=framework,
+            args_parser=args_parser,
+            device=device,
+            binary_output=True,
+            task=task,
+        )
+        self.stop_token = stop_token
+        self.padding_text = padding_text if padding_text else self.PADDING_TEXT
+        self.xlm_language = xlm_language
+        self.length = length
+        self.temperature = temperature
+        self.top_k = top_k
+        self.top_p = top_p
+        self.repetition_penalty = repetition_penalty
+        self.do_sample = do_sample
+        self.num_return_sequences = num_return_sequences
+        self.model_type = self.model.config.to_dict()["model_type"]
+        self.preprocessing_functions = {
+            "ctrl": self.prepare_ctrl_input,
+            "xlm": self.prepare_xlm_input,
+            "xlnet": self.prepare_xlnet_input,
+            "transfo-xl": self.prepare_transfoxl_input,
+        }
+
+    def __call__(self, *texts, **kwargs):
+        inputs = self._args_parser(*texts, **kwargs)
+
+        all_generated_sequences = []
+        for i, prompt_text in enumerate(inputs):
+            # Manage correct placement of the tensors
+            with self.device_placement():
+                tokens = self.prepare_input(prompt_text)
```

Comment:
For now let's use a "dirty" `if self.model.__class__ .__name__ in ['XLNetLMHeadModel', 'TransfoXLLMHeadModel']: prompt_text = PADDING_TEXT + prompt_text`
Then we should do the encoding in the line after that:
`            encoded_prompt = self.tokenizer.encode(
                prompt_text, add_special_tokens=False, return_tensors=self.framework
            )`

Reply:
do you think, we should set `PADDING_TEXT` outside of the class? Otherwise, should be `self.PADDING_TEXT`. Will set to the latter for now.